File: benchmarks/core-comparison/programs/builtin-reduction.c
#include <stdio.h>

int main(int argc, char **argv) {
    (void) argv;
    double values[{{COUNT}}];
    for (int i = 0; i < {{COUNT}}; ++i) {
        values[i] = (double) (i + argc);
    }
    double sum = 0.0;
    for (int i = 0; i < {{COUNT}}; ++i) {
        sum += values[i];
    }
    printf("%.17g\n", sum + sum / (double) {{COUNT}} + (double) {{COUNT}});
    return 0;
}

File: benchmarks/core-comparison/programs/builtin-reduction.rs
fn main() {
    let start = std::hint::black_box(1.0_f64);
    let mut values = [0.0_f64; {{COUNT}}];
    for (index, value) in values.iter_mut().enumerate() {
        *value = index as f64 + start;
    }
    let sum: f64 = values.iter().sum();
    println!("{:.17}", sum + sum / {{COUNT}}.0 + {{COUNT}}.0);
}
